sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/measure/MeasureCache.java

```diff
@@ -23,62 +23,52 @@
 import javax.annotation.CheckForNull;
 import org.sonar.api.batch.ScannerSide;
 import org.sonar.api.batch.measure.MetricFinder;
-import org.sonar.api.measures.Measure;
-import org.sonar.api.resources.Resource;
+import org.sonar.api.batch.sensor.measure.internal.DefaultMeasure;
 import org.sonar.scanner.index.Cache;
-import org.sonar.scanner.index.Caches;
 import org.sonar.scanner.index.Cache.Entry;
+import org.sonar.scanner.index.Caches;
 
 /**
  * Cache of all measures. This cache is shared amongst all project modules.
  */
 @ScannerSide
 public class MeasureCache {
 
-  private final Cache<Measure> cache;
+  private final Cache<DefaultMeasure<?>> cache;
 
   public MeasureCache(Caches caches, MetricFinder metricFinder) {
-    caches.registerValueCoder(Measure.class, new MeasureValueCoder(metricFinder));
+    caches.registerValueCoder(DefaultMeasure.class, new MeasureValueCoder(metricFinder));
     cache = caches.createCache("measures");
   }
 
-  public Iterable<Entry<Measure>> entries() {
+  public Iterable<Entry<DefaultMeasure<?>>> entries() {
     return cache.entries();
   }
 
-  public Iterable<Measure> all() {
+  public Iterable<DefaultMeasure<?>> all() {
     return cache.values();
   }
 
-  public Iterable<Measure> byResource(Resource r) {
-    return byComponentKey(r.getEffectiveKey());
-  }
-
-  public Iterable<Measure> byComponentKey(String effectiveKey) {
+  public Iterable<DefaultMeasure<?>> byComponentKey(String effectiveKey) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1452)